gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java

```diff
@@ -134,9 +134,41 @@ protected List<FileStatus> getFilesAtPath(FileSystem fs, Path path, PathFilter f
     return recursivelyGetFilesAtDatePath(fs, path, "", fileFilter, 1, startDate, endDate, formatter);
   }
 
+  /**
+   * Checks if the datePath provided is in the range of the start and end dates.
+   * Rounds startDate and endDate to the same granularity as datePath prior to comparing.
+   * Returns true if the datePath provided is in the range of start and end dates, inclusive.
+   * @param startDate
+   * @param endDate
+   * @param datePath
+   * @return true/false
+   */
+  public Boolean checkPathDateTimeValidity(LocalDateTime startDate, LocalDateTime endDate, String datePath) {
+    boolean onlyNumbers = datePath.matches("^[0-9/]+$");
+    if (onlyNumbers) {
+      String allGranularity = "yyyy/MM/dd/HH/mm/ss/SSS";
+      String roundGranularity = allGranularity.substring(0, datePath.length());
+      DateTimeFormatter formatGranularity = DateTimeFormat.forPattern(roundGranularity);
+
+      LocalDateTime traversedDatePathRound = formatGranularity.parseLocalDateTime(datePath);
+      LocalDateTime startDateRound = formatGranularity.parseLocalDateTime(startDate.toString(roundGranularity));
+      LocalDateTime endDateRound = formatGranularity.parseLocalDateTime(endDate.toString(roundGranularity));
+
+      boolean afterOrOnStartDate = traversedDatePathRound.isAfter(startDateRound) || traversedDatePathRound.isEqual(startDateRound);
+      boolean beforeOrOnEndDate = traversedDatePathRound.isBefore(endDateRound) || traversedDatePathRound.isEqual(endDateRound);
+      return afterOrOnStartDate && beforeOrOnEndDate;
+    }
+    return false;
```

Comment:
Should it return true?

Reply:
It should return false because if the datePath has non-numerical characters (excluding `/`), then it should not keep traversing as it is not in the right dateTime format